concurrent.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Concurrent server using gevent
+"""
+
+import server
+from gevent import socket as g
+
+
+def return_concurrent_request():
+    ADDR = ('127.0.0.1', 8000)
+    socket = g.socket(
+        g.AF_INET, g.SOCK_STREAM, g.IPPROTO_IP
+    )
+
+    socket.bind(ADDR)
+    socket.listen(5)
+    while True:
+        try:
+            conn, addr = socket.accept()
+            request = ""
+            while True:
+                msg = conn.recv(1024)
+                request += msg
+                if len(msg) < 1024:
+                    break
+            try:
+                response = server.response_ok(server.parse_request(request))
```

Comment:
chaining the calls like this is totally okay.  However, I'd take it a step farther, in that now `response_ok` is doing two tasks: building a response and resolving and reading disk resources. If you re-established the separation of `response_ok` and `resolve_uri`, you could make a call chain like this:

```python
response = server.response_ok(server.resolve_uri(server.parse_request(request)))
```

That's functional programming.  Each function takes in a well-defined input and returns a well-defined output to be used by the next function in the chain.